kernel/src/returncode.rs

```diff
@@ -40,3 +40,23 @@ impl From<ReturnCode> for isize {
         }
     }
 }
+
+impl From<ReturnCode> for usize {
+    fn from(original: ReturnCode) -> usize {
```

Comment:
It's a little weird to me that going from a ReturnCode to a `isize` and a `usize` will generate a different value, i.e.
```rust
let r = ReturnCode::FAIL;
let i: isize = r::into();
let u: usize = r::into();
assert(i as usize == u) // fails
```

I think you can fix this and also eliminate the duplicated match list by changing this implementation to:

```rust
impl From<ReturnCode> for usize {
    fn from(original: ReturnCode) -> usize {
        isize::from(original) as usize
    }
}
```